config/always-compiler-options.m4

```diff
@@ -100,7 +121,7 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_CC_NO_BOOL_COMPARE], [
 	AC_MSG_CHECKING([whether $CC supports -Wno-bool-compare])
 
 	saved_flags="$CFLAGS"
-	CFLAGS="$CFLAGS -Wbool-compare"
+	CFLAGS="$CFLAGS -Werror -Wbool-compare"
```

Comment:
`-Wall` is ok. However `-Werror` is the maintainer's nightmare as it breaks builds after upgrading to a new compiler with new checks.

Reply:
It is needed to check support for the warning in clang. Otherwise clang emits a warning about the unknown warning flag where gcc would have exited. `-Werror=unknown-warning-option -Wbool-compare` works to make that an error in clang but gcc doesn't know unknown-warning-option so the check would be broken for gcc. `-Werror -Wbool-compare` works in both clang and gcc.

Reply:
Is it not possible to detect what compiler flags are supported and then set them?

Reply:
That's what this is doing, detecting compiler flags.

Reply:
Sorry for my unqualified review :) I should have checked the context around this file.

Reply:
Thanks for taking a look though!